File: snuggle_server/snuggle_server/src/main/java/com/ssafy/snuggle/model/service/FavoriteService.java
package com.ssafy.snuggle.model.service;

import java.util.List;

import com.ssafy.snuggle.model.dto.Favorite;
import com.ssafy.snuggle.model.dto.Tagging;

public interface FavoriteService {

	 /**
     * Favorite를 등록한다.
     * @param Favorite
     */
    int addFavorite(Favorite favorite);
    

    /**
     * 사용자가 좋아요 한 favorite 조회
     * @param userId
     */
    List<Favorite> getUserFavorites(String userId);
    
    
    /**
     * 사용자가 좋아요 한 tagging 리스트 조회
     * @param userId
     */
    List<Tagging> getFavoriteTaggingList(String userId);
    
}

File: snuggle_final_app/app/src/main/java/com/ssafy/snuggle_final_app/data/service/FavoriteService.kt
package com.ssafy.snuggle_final_app.data.service

import com.ssafy.snuggle_final_app.data.model.dto.Favorite
import com.ssafy.snuggle_final_app.data.model.dto.FavoriteRequest
import com.ssafy.snuggle_final_app.data.model.dto.Tagging
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface FavoriteService {

    // 즐겨찾기 추가
    @POST("snuggle/favorite/insert")
    suspend fun toggleFavorite(@Body favorite: FavoriteRequest): Response<Boolean>

    // 사용자 즐겨찾기 조회
    @GET("snuggle/favorite/info")
    suspend fun getFavoriteTaggingList(@Query("userId") userId: String): Response<List<Tagging>>

    @GET("snuggle/favorite/{userId}")
    suspend fun getUserFavorites(@Path("userId") userId: String): Response<List<Favorite>>
}
